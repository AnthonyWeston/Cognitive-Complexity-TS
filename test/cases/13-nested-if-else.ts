function f() {
    if (true) {
        if (true) {
        } else if (true) {
        } else {
        }
    }
}

function g() {
    if (true) { // +1
        // depth 1
        if (true) { // +1 inherent, +1 for depth
            // depth 2
            while (true); // +1 +2
        } else if (true) { // +1 inherent, +0 despite depth
            // depth 2
            while (true); // +1 +2
        } else { // +1 inherent, +0 despite depth
            // depth 2
            while (true); // +1 +2
        }
    }
}
