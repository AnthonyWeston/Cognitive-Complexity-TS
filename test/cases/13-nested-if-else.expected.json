{
    "13-nested-if-else.ts": {
        "score": 19,
        "inner": [
            {
                "name": "f",
                "score": 5,
                "line": 1,
                "column": 1
            },
            {
                "name": "g",
                "score": 14,
                "line": 10,
                "column": 1
            }
        ]
    }
}
